fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/FDBInQueryTest.java

```diff
@@ -103,14 +126,18 @@
     @Test
     public void testInQueryNoIndex() throws Exception {
         complexQuerySetup(NO_HOOK);
+        final QueryComponent filter = Query.field("num_value_2").in(asList(0, 2));
         RecordQuery query = RecordQuery.newBuilder()
                 .setRecordType("MySimpleRecord")
-                .setFilter(Query.field("num_value_2").in(asList(0, 2)))
+                .setFilter(filter)
                 .build();
 
         // Scan(<,>) | [MySimpleRecord] | num_value_2 IN [0, 2]
         RecordQueryPlan plan = planner.plan(query);
-        assertThat(plan, filter(query.getFilter(), descendant(scan(unbounded()))));
+        
+        assertTrue(filterPlan(descendantPlans(scanPlan().where(scanComparisons(ScanComparisons.unbounded()))))
+                .where(queryComponents(exactly(equalsObject(filter)))).matchesExactly(plan));
```

Comment:
Just like the previous matchers, this will not be very helpful in establishing where something failed. Even before the failure for a match was not displayed in a helpful way. It just claimed that the plan (stringified) did not match what the matcher was going to match. But there is naturally no point of deviation where things go wring inside the structure that tell you that a sub plan is different as a lot of matching is done on existential predication. The matchers do not have a proper `toString()` but honestly I have never found much use in that.

Reply:
I added another wrapper that looks like:
```assertMatchesExactly(plan,
                    typeFilterPlan(
                            scanPlan().where(scanComparisons(unbounded())))
                            .where(recordTypes(containsAll(ImmutableSet.of("MySimpleRecord")))));```

Reply:
Again, currently it will only bomb with a true vs. false and you will have to navigate using the provided stack.